client/finality-grandpa/src/communication/gossip.rs

```diff
@@ -539,21 +542,26 @@ impl<N: Ord> Peers<N> {
 	}
 
 	fn authorities(&self) -> usize {
-		self.inner.iter().filter(|(_, info)| info.roles.is_authority()).count()
+		self.inner.iter().filter(|(_, info)| matches!(info.roles, ObservedRole::Authority)).count()
```

Comment:
What about `OurGuardedAuthority`?

Reply:
Purposefully excluded since we always broadcast to them.